### PR TITLE
chore(chain)!: Rename `Append` to `Merge`

### LINKED ISSUE #1467: Rename `Append` trait to `Merge`
I think this captures better what this is doing. I think it was originally named append since the underlying this always used the core append method but we stopped using that because it's not efficient anyway.

### PR DESCRIPTION
**Pull request by LagginTimes**
Fixes #1467.

### Description

Renames the `Append` trait to `Merge`.

### Changelog notice

* Rename `bdk_chain::Append` to `bdk_chain::Merge`.

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

**Review by oleonardolima — COMMENTED** [review]
utACK 4a12aac0c0897fc95e6f32fde4885e2399cfd46b

File: crates/chain/tests/test_keychain_txout_index.rs [hunk 1]
```diff
@@ -89,7 +89,7 @@ fn append_changesets_check_last_revealed() {
         keychains_added: BTreeMap::<(), _>::new(),
         last_revealed: rhs_di,
     };
-    lhs.append(rhs);
+    lhs.merge(rhs);
```
Comment (oleonardolima) [inline crates/chain/tests/test_keychain_txout_index.rs:91]: @LagginTimes nit: should we rename the test fn name ? wdyt ?

**Review by oleonardolima — APPROVED** [review]
ACK 4a12aac0c0897fc95e6f32fde4885e2399cfd46b

File: crates/chain/tests/test_keychain_txout_index.rs (hunk 1 above)
Comment (notmandatory) [reply crates/chain/tests/test_keychain_txout_index.rs:91]: A few spots in this file to rename append -> merge and then should be ready to go.
line 55: in the docs
line 61: test name
line 683: function wasn't renamed, don't know why not caught by CI

**Review by oleonardolima — APPROVED** [review]
ACK 7b948d5fa92c79450a45d6d227cdebef92d9cdd3

**Review by notmandatory — APPROVED** [review]
ACK 962305f41546613b5b41b4131d4a5594f01a2de0

**Pull request merged**